pkg/pool-stable/test/ComposableStablePool.test.ts

```diff
@@ -1575,31 +1597,107 @@ describe('ComposableStablePool', () => {
 
             const rate = await pool.getRate();
 
-            expect(rate).to.be.equalWithError(expectedRate, 0.0001);
+            expect(rate).to.be.almostEqual(expectedRate, 0.0001);
           });
         });
 
         context('with protocol fees', () => {
-          sharedBeforeEach('swap bpt in', async () => {
-            const amount = fp(50);
-            const tokenIn = pool.bpt;
-            const tokenOut = tokens.second;
+          let feeAmount: BigNumber; // The number of tokens in the Pool that are fees
 
-            await tokens.mint({ to: lp, amount });
-            await tokens.approve({ from: lp, to: pool.vault });
+          function itReportsRateCorrectly() {
+            it('takes into account unminted protocol fees', async () => {
+              const invariant = await pool.estimateInvariant();
+
+              // The virtual supply does not include the unminted protocol fees. We need to adjust it by computing those.
+              // Since all balances are relatively close and the pool is balanced, we can estimate their prices as being
+              // equal to then price the due protocool fees.
```

Comment:
```suggestion
              // Since all balances are relatively close and the pool is balanced, we can estimate their prices as being
              // equal to then price the due protocool fees.
```
              // Since all balances are relatively close and the pool is balanced, we can simply add the fee amount to
              // the current balances to obtain the final sum.